### ISSUE TITLE
Retrieve system data from the server of the site

### ISSUE THREAD

**Issue by carcam** [akeeba/panopticon#307]
## Is your feature request related to a problem? Please describe.
Sometimes, due to failed backups or logs constantly growing, your site gets out of space.

## Describe the solution you'd like
I would like a monitor on free disk space to detect when we are running issues

## Describe alternatives you've considered
A bash script that checks disk space and sends a daily email with it.

## Additional context
We discussed this in the Small Business Owners group ;)

**Comment by nikosdion** [comment]
I can confirm I am currently working on it.

Looking into this, I decided to report the disk free from two places (if possible): the mountpoint where the site's root is, and the mountpoint where the database files are located in. Caveat: if the db server is remote you get no insight. **REASONING**: I have seen at least two servers the last 4 months which had oodles of free space in the web root, but ran out of space in the DB mount, causing the site to crash (thankfully, both sites were rescued, one by yours truly, the other by the client's host).

I am also reporting the memory, if we can read that info, because memory pressure can be a real pain in the butt. What good does it make having loads of disk space if memory is limited and causes swap thrashing?

On the same vein, I am trying to collect CPU usage information including `iowait` (how much time the kernel has been sitting on its butt, waiting for I/O to complete). A high `iowait` is a strong indication that your server is suffering under load and might spin out of control.

Remember, these are just _points in time_, collected every 15' with the rest of the update information. If you want to monitor these values over time I strongly recommend using HetrixTools or a similar site uptime and **server** monitoring service. I am just recommending what I use myself.

**Comment by brianteeman** [comment]
have you seen https://github.com/bitts/zabbix_joomla

**Comment by nikosdion** [comment]
@brianteeman No, but it falls in the same category as https://github.com/akeeba/panopticon/wiki/Uptime-Monitoring#why-not-integrate-with-a-third-party-service  As I have explained there, I am not going to start implementing integrations with any third party services.

The idea of this feature is to provide a point-in-time view of critical server parameters. It is not meant to monitor the server. If you start having problems with the server it makes sense to invest in a **server** monitoring service and set alerts. I've already documented I am using HetrixTools.

**Comment by carcam** [comment]
Thank you both!! I will definitely check Zabbix and HetrixTools. In any case, @nikosdion your plan for this looks promising so I'll stay tuned ;)

**Comment by brianteeman** [comment]
@nikosdion I mentioned it because it seemed to do what you said you were implementing

**Comment by brianteeman** [comment]
thanks
